core/schema/modulesource.go

```diff
@@ -1,1648 +1,2031 @@
 package schema
 
 import (
+	"bytes"
 	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
+	"net/url"
+	"os"
+	"path"
 	"path/filepath"
 	"sort"
 	"strings"
 
 	"dagger.io/dagger/telemetry"
-	"golang.org/x/sync/errgroup"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
 	"github.com/dagger/dagger/core"
 	"github.com/dagger/dagger/core/modules"
 	"github.com/dagger/dagger/dagql"
+	"github.com/dagger/dagger/engine"
 	"github.com/dagger/dagger/engine/buildkit"
 	"github.com/dagger/dagger/engine/client/pathutil"
 	"github.com/dagger/dagger/engine/slog"
 	"github.com/dagger/dagger/engine/vcs"
 	"github.com/go-git/go-git/v5/plumbing/transport"
-	"github.com/tonistiigi/fsutil/types"
+	"github.com/moby/buildkit/util/bklog"
+	"github.com/opencontainers/go-digest"
+	fsutiltypes "github.com/tonistiigi/fsutil/types"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 type moduleSourceArgs struct {
 	// avoiding name "ref" due to that being a reserved word in some SDKs (e.g. Rust)
 	RefString string
 	RefPin    string `default:""`
+	Stable    bool   `default:"false"`
```

Comment:
I *think* while we're here we can remove this. https://github.com/dagger/dagger/pull/8587 added pins, and the logic of pins essentially means that everything is pretty much stable now.